enumerables.rb

```diff
@@ -0,0 +1,156 @@
+module Enumerable
+  # my_each
+  def my_each()
+    return to_enum(:my_each) unless block_given?
+
+    arr = to_a if self.class == Range
+    arr = self if self.class == Array
+    arr = to_a if self.class == Hash
+    length.times do |n|
+      yield(arr[n])
+    end
+    self
+  end
```

Comment:
`my_each` method throws an error when iterated over a range of values.

-  `each` for range values

```
a = (0..3)
a.each {|x| print x}
result: 0123
```

-  `my_each` for range values

```
a = (0..3)
a.my_each {|x| print x}
result: enumerables.rb:7:in `my_each': undefined local variable or method `length' for 0..3:Range (NameError)
```